src/OpenLoco/Windows/CompanyWindow.cpp

```diff
@@ -1384,11 +1384,17 @@ namespace OpenLoco::Ui::Windows::CompanyWindow
                 case main_colour_aircraft:
                 case main_colour_ships:
                 {
-                    registers regs;
-                    regs.edx = widgetIndex;
-                    regs.esi = (int32_t)self;
-                    regs.edi = (int32_t)&self->widgets[widgetIndex];
-                    call(0x00433119, regs);
+                    auto* company = CompanyManager::get(self->number);
+                    Colour_t selectedColour;
+                    if (widgetIndex != main_colour_scheme)
+                    {
+                        auto vehicleType = widgetIndex - main_colour_steam_locomotives;
+                        selectedColour = company->vehicleColours[vehicleType].primary;
+                    }
+                    else
+                        selectedColour = company->mainColours.primary;
+
+                    Dropdown::showColour(self, &self->widgets[widgetIndex], 0x7FFFFFFF, selectedColour, self->getColour(WindowColour::secondary));
```

Comment:
This still needs to have masks applied to it to make competitor's colours unavailable. Only for the primary colour.